shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-dialect/shardingsphere-data-pipeline-postgresql/src/main/java/org/apache/shardingsphere/data/pipeline/postgresql/check/datasource/PostgreSQLDataSourceChecker.java

```diff
@@ -54,9 +54,11 @@ private void checkPrivilege(final DataSource dataSource) {
                     throw new PipelineJobPrepareFailedException(String.format("No role exists, rolname: %s.", metaData.getUserName()));
                 }
                 String isSuperRole = resultSet.getString("rolsuper");
+                // compatible openGauss database, role name is SYSADMIN;
+                String isSystemAdminRole = resultSet.getString("rolsystemadmin");
                 String isReplicationRole = resultSet.getString("rolreplication");
-                log.info("checkPrivilege: isSuperRole: {}, isReplicationRole: {}", isSuperRole, isReplicationRole);
-                if (StringUtils.equalsIgnoreCase(isSuperRole, "f") && StringUtils.equalsIgnoreCase(isReplicationRole, "f")) {
+                log.info("checkPrivilege: isSuperRole: {}, isReplicationRole: {}, isSystemAdminRole:{}", isSuperRole, isReplicationRole, isSystemAdminRole);
```

Comment:
Is it just available in openGauss? If it's, could we create a openGauss impl of DataSourceChecker?